includes/class-wc-payments.php

```diff
@@ -466,6 +466,12 @@ public static function init() {
 			WC_Payments_Subscriptions::init( self::$api_client, self::$customer_service, self::get_gateway(), self::$account );
 		}
 
+		if ( WC_Payments_Features::is_express_checkout_enabled() ) {
+			add_action( 'woocommerce_after_add_to_cart_quantity', [ __CLASS__, 'display_express_checkout_button_separator_html' ], -1 );
+			add_action( 'woocommerce_proceed_to_checkout', [ __CLASS__, 'display_express_checkout_button_separator_html' ], -1 );
+			add_action( 'woocommerce_checkout_before_customer_details', [ __CLASS__, 'display_express_checkout_button_separator_html' ], -1 );
```

Comment:
After removing the additional separator coming from the Payment Request button on the Cart, Checkout, and Product Details pages, one might be surprised that the separator disappeared, but the button itself is still rendered _**after**_ the `--OR--` separator we have in place, and not before it. 

<p align="center">
<img width=60% height=60% alt="image" src="https://user-images.githubusercontent.com/22319444/212651658-ebceb426-a6fc-469c-80b8-83d3bd7ca50d.png">
</p>

This is because the separator rendering callbacks are hooked here with the higher priority `-1` than the Payment Request button (`1` prio [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-payment-request-button-handler.php#L76), [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-payment-request-button-handler.php#L79) and also [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-payment-request-button-handler.php#L82)). 

We'd need to align the priority of the Payment Request button rendering with other buttons - I can see that both WooPay and Stripe Link have `-2` priority, so maybe this value should be our choice.

> 💡 Note: I think that the sequence of buttons rendered might be important from the business perspective. For instance, we might want to always render the WooPay Express Checkout button as the first one, then Stripe Link, and then the Payment Request button. Unfortunately, the sequence is unknown to me, but I think we can consult someone from the product or UX team to confirm this. That being said, if we set the callback hook priority for the Payment Request button to be `-2`, we'll get the following state, and I'm not sure if the `GPay -> WooPay -> Stripe Link` is the desired sequence. 

<p align="center">
<img width=60% height=60% alt="image" src="https://user-images.githubusercontent.com/22319444/212656906-4cba4636-b86b-4e98-910b-2777dc590cd5.png">
</p>

Reply:
Ah, I didn't notice this previously. It makes sense that there might be a preferred order for express checkout buttons. I'll apply the `-2` priority for now but will look into this and follow up with a new issue if needed.